viz/cmd/allow-scrapes.go

```diff
@@ -0,0 +1,143 @@
+    - name: proxy-admin
+      kind: Server
+      group: policy.linkerd.io
+  rules:
+    - matches:
+      - path:
+          value: "/metrics"
+---
+apiVersion: policy.linkerd.io/v1alpha1
+kind: HTTPRoute
+metadata:
+  name: proxy-probes{{ if .TargetNs }}\n  namespace: {{ .TargetNs }}{{end}}
+  annotations:
+    linkerd-io/created-by: {{ .ChartName }} {{ .Version }}
+  labels:
+    linkerd.io/extension: {{ .ExtensionName }}
+spec:
+  parentRefs:
+    - name: proxy-admin
+      kind: Server
+      group: policy.linkerd.io
+  rules:
+    - matches:
+      - path:
+          value: "/live"
+      - path:
+          value: "/ready"
+---
+apiVersion: policy.linkerd.io/v1alpha1
+kind: AuthorizationPolicy
+metadata:
+  name: prometheus-scrape{{ if .TargetNs }}\n  namespace: {{ .TargetNs }}{{end}}
+  annotations:
+    linkerd-io/created-by: {{ .ChartName }} {{ .Version }}
+  labels:
+    linkerd.io/extension: {{ .ExtensionName }}
+spec:
+  targetRef:
+    group: policy.linkerd.io
+    kind: HTTPRoute
+    name: proxy-metrics
+  requiredAuthenticationRefs:
+    - kind: ServiceAccount
+      name: prometheus
+      namespace: {{ .VizNs }}
+---
+apiVersion: policy.linkerd.io/v1alpha1
+kind: AuthorizationPolicy
+metadata:
+  name: proxy-probes{{ if .TargetNs }}\n  namespace: {{ .TargetNs }}{{end}}
+  annotations:
+    linkerd-io/created-by: {{ .ChartName }} {{ .Version }}
+  labels:
+    linkerd.io/extension: {{ .ExtensionName }}
+spec:
+  targetRef:
+    group: policy.linkerd.io
+    kind: HTTPRoute
+    name: proxy-probes
+  requiredAuthenticationRefs:
+    - kind: NetworkAuthentication
+      group: policy.linkerd.io
+      name: kubelet
+      namespace: {{ .VizNs }}`
+)
+
+type templateOptions struct {
+	ChartName     string
+	Version       string
+	ExtensionName string
+	VizNs         string
+	TargetNs      string
+}
+
+// newCmdAllowScrapes creates a new cobra command `allow-scrapes`
+func newCmdAllowScrapes() *cobra.Command {
+	options := templateOptions{
+		ExtensionName: ExtensionName,
+		ChartName:     vizChartName,
+		Version:       version.Version,
+		VizNs:         defaultNamespace,
+	}
+	cmd := &cobra.Command{
+		Use:   "allow-scrapes [flags]",
+		Short: "Output Kubernetes resources to authorize Prometheus scrapes",
+		Long:  `Output Kubernetes resources to authorize Prometheus scrapes in a namespace or cluster with config.linkerd.io/default-inbound-policy: deny.`,
+		Example: `# Allow scrapes in the default namespace
+linkerd viz allow-scrapes | kubectl apply -f -
+
+# Allow scrapes in the 'emojivoto' namespace
+linkerd viz allow-scrapes --namespace emojivoto | kubectl apply -f -`,
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			t := template.Must(template.New("allow-scrapes").Parse(allowScrapePolicy))
+			return t.Execute(os.Stdout, options)
+		},
+	}
+	cmd.Flags().StringVarP(&options.TargetNs, "namespace", "n", options.TargetNs, "If set, the generated resources will target this namespace.")
```

Comment:
I'd consider making the namespace flag mandatory.  If omitted from both the linkerd and kubectl commands, it will put the resources into the default namespace, which seems like a gotcha.  Especially because the `linkerd viz check --proxy` without a namespace flag will check all namespaces.  Better to make the user be explicit here imo.